### ISSUE TITLE
@wordpress/apiFetch fails remote site requests.

### ISSUE THREAD

**Issue by hansschuijff** [WordPress/gutenberg#46787]
### Description

Not sure if this is a bug or a feature, but I've been building a remote tickets plugin to use the REST api to get event ticket info from one of my websites and present it on another (I use event tickets from Tribe). 

To build this I needed to do a cross site request to a remote (wordpress) website and I can do that easily using php in a dynamic block on the frontend, or by using either window.fetch, XMLHTTPrequest or axios in edit.js.

But I don't seem to be able to get it working using apiFetch. I've read the docs and it says I can use the url: option to pass a complete url like I can in window.fetch and it makes the request, but it adds a nonce and that fails the request, so I don't get the resulting tickets info.

The problem seems to be (when I look into the code) that apiFetch is hardwired to add a x-wp-nonce to the header and that only works on samesite requests. It gets the nonce by first requesting  `https:/{samesite domain}/wp-admin/admin-ajax.php?action=rest-nonce`. See also discussion #46780.

So if I'm not missing anything, I can either only use apiFetch for samesite requests or I need a way to prevent apiFetch from adding a nonce to a remote URL (of it should automatically not do that when the domain is not the local domain).

if it's a feature, the bug is that the developer docs don't state that apiFetch is limited to only samesite requests and will fail on remote site requests. That prevents me and other spending days on trying to find out why it doesn't work and other methods are just fine with such requests. 

Right now the docs just suggests that you can use middleware or pass the url option to change the rootURL, but doesn't tell it should always be a URL pointing to the same site or else it will fail.

If it's an unintended feature, the limitation should perhaps be lifted, probably by not adding a nonce to requests that target other sites than the WordPress install the block is in.

### Step-by-step reproduction instructions

1. Add the following code the src/edit.js file of a block plugin (I used @wordpress/create-block to generate one) and activate it.

the code below is what I used, but the url could be to any other remote domain:
```
import apiFetch from '@wordpress/api-fetch';
apiFetch( {
	url: 'https://dewitteprins.nl/wp-json/wp/v2/posts',
	method: 'GET',
} ).then( ( res ) => {
	console.log( "apiFetch result: ", res );
} ).catch( ( err ) => {
	console.log( "apiFetch error: ", err );
} );

```
2. add the block to a post.

3. See the resulting error (403) in the browsers console.log.

### Environment info

- WordPress version 6.1.1
- Firefox dev ed 109.0b6 (64-bits)
- Windows 11
- Gutenberg plugin not active.
- I ran my local dev website on on Local v.6.6.0+6231

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
This issue may be related to #13491.

Also, I don't know the background behind the introduction of this package (formerly `apiRequest`), but from [this comment](https://github.com/WordPress/gutenberg/pull/7566#discussion_r202213855) It might be intended for use with the WP REST API.

> I think parsing error response bodies makes sense since this package is specifically designed to work with the WP REST API, which we know returns nicely formatted error responses:

**Comment by hansschuijff** [comment]
@t-hamano thank you! 

You found some relevant discussions on the topic, thanks for that. I understand now apiFetch is primarily build to be used on WordPress websites and then specifically samesite websites, but that is not stated clearly in the docs and has cost me lots of time by now.

But if it's so, why not add something in the package that blocks unintended the use more clearly anyway? 

And if the windows.fetch alternative vanilla code is so simple, why doesn't it just handle that too? It seems such a small thing to make it either clear for dev that it is samesite only, or just support it in a minimalistic way.

In the discussions you found, someone called apiFetch the goto for Gutenberg block devs. And perhaps it should be, but then it should provide also for blocks that are build to use outside api's and non WP api's. Otherwise why not just use window.fetch all the time?

In the end it's not clear and the docs mainly seem to present it as being more or less the same functionality as window.fetch with the advantage of getting the json directly back from the fetch. Without having any other data, without having the advantage of building gutenberg and knowing the discussions about it, devs like me think based on the docs they can use it as such, and perhaps we should. 

There is a reason one stop shopping is populair. It makes live easier for customers.

**Comment by helgatheviking** [comment]
Can confirm that I have also lost a lot of time wondering about that (especially challenging when I was trying to get a basic example set up). Without that ability, what is the advantage of having this over pure `fetch`? And at minimum, should be documented that it's not supported at all.
